### PR TITLE
main: add initial implementation

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			LocalIPv4:  local,
+			Hostname:   hostname,
+		}, nil
+	default:
+		panic("bad provider")
+	}
+}
+
+func write_metadata(out *os.File, metadata metadata) error {
+	if _, err := out.Write([]byte(fmt.Sprintf("COREOS_IPV4_PUBLIC=%s\n", metadata.PublicIPv4))); err != nil {
```
Comment (marineam) [inline main.go]: Write is silly because: http://golang.org/pkg/os/#File.WriteString
But then Sprintf is silly because: http://golang.org/pkg/fmt/#Fprintf

Comment (marineam) [inline main.go]: Also, writing out a `net.IP` is fine for now but how is this going to look for providers that have different sets of addresses they define? Writing out the zero value `0.0.0.0` isn't going to be useful.

Comment (crawford) [inline main.go]: If only there was a way to determine if a value was a zero value.

Comment (crawford) [inline main.go]: Updated.

Comment (marineam) [inline main.go]: You mean like `ip.Equal(net.IPv4zero)` Mr. Snarky? :-P

**Comment by marineam** [comment]
LGTM

File: main.go (hunk 1 above)
Comment (crawford) [inline main.go]: No, a general `IsZero` function. That way I don't need those two helpers.

Comment (marineam) [inline main.go]: Oops, actually, I mislead you because IP is a slice so the zero value is nil, not IPv4zero

Comment (marineam) [inline main.go]: Meaning you would still wind up with `FOO_IPV4=<nil>` with the code as-is for non ec2 things in the future.

Comment (crawford) [inline main.go]: Shit. You're right.

**Pull request merged**